torch/testing/_internal/common_methods_invocations.py

```diff
@@ -14356,11 +14357,19 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
                          device_type='cuda', dtypes=(torch.bfloat16,), active_if=not SM80OrLater),
             DecorateInfo(unittest.skip("Skipped!"), 'TestMeta', 'test_dispatch_symbolic_meta_outplace',
                          device_type='cuda', dtypes=(torch.bfloat16,), active_if=not SM80OrLater),
-            # registered in fake_impls.py instead of _meta_registrations.py
+            # registered in fake_impls.py instead of _meta_registrations.py, so meta kernels will fail.
+            # However, for implementations that fall back to the constituent ops, the meta kernels may not
+            # fail. Fused kernels will fail, whereas unfused kernels will not fail.
+            # All fused kernels support bf16 and fp16 - so if fused attention is supported, the test will fail.
+            # mem_eff_attention also supports fp32 - so if it is supported the test will fail.
             DecorateInfo(unittest.expectedFailure, "TestMeta", "test_dispatch_meta_outplace",
-                         dtypes=(torch.bfloat16, torch.float16, torch.float32)),
+                         dtypes=(torch.bfloat16, torch.float16), active_if=PLATFORM_SUPPORTS_FUSED_ATTENTION),
```

Comment:
So right now the:
PLATFORM_SUPPORTS_FUSED_ATTENTION is a superset of the two:
PLATFORM_S_FLASH.., PLATFORM_S_EFFIC..

It just so happends that, `PLATFORM_S_EFFIC = PLATFORM_SUPPORTS_FUSED_ATTENTION`

since if a platforms supports it also supports mem_eff. 

Do you want the, bfloat16/float16 skips to be gated by flash_attention?

Reply:
AFAIK mem_eff supports all 3 of {bf16, fp16, fp32}, right? I.e. any fused attention kernel will make this test fail with {bf16, fp16}, so we should xfail {bf16, fp16} as long as any fused attention kernel is supported

Reply:
Yes to the first question and you are right